sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,773 @@
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifOtpCtrlConfigLocked,
+} dif_otp_ctrl_config_result_t;
+
+/**
+ * The result of a Direct Access Interface (DAI) operation.
+ */
+typedef enum dif_otp_ctrl_dai_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifOtpCtrlDaiOk = kDifOtpCtrlOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifOtpCtrlDaiError = kDifOtpCtrlError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifOtpCtrlDaiBadArg = kDifOtpCtrlBadArg,
+  /**
+   * Indicates that the DAI is busy and can't accept another incomming command.
+   */
+  kDifOtpCtrlDaiBusy,
+  /**
+   * Indicates that the given partition does not support this operation.
+   */
+  kDifOtpCtrlDaiBadPartition,
+  /**
+   * Indicates that the attempted DAI operation would go out of range of the
+   * requested partition.
+   */
+  kDifOtpCtrlDaiOutOfRange,
+  /**
+   * Indicates that the given address was not correctly aligned.
+   */
+  kDifOtpCtrlDaiUnaligned,
+} dif_otp_ctrl_dai_result_t;
+
+/**
+ * An OTP interrupt request type.
+ */
+typedef enum dif_otp_ctrl_irq {
+  /**
+   * Indicates that an asynchronous transaction completed.
+   */
+  kDifOtpCtrlIrqDone,
+  /**
+   * Indicates that an error has occured in the OTP controller.
+   */
+  kDifOtpCtrlIrqError,
+  // Your IRQs here!
+} dif_otp_ctrl_irq_t;
+
+/**
+ * A snapshot of the enablement state of the interrupts for OTP.
+ *
+ * This is an opaque type, to be used with the `dif_otp_ctrl_irq_disable_all()`
+ * and
+ * `dif_otp_ctrl_irq_restore_all()` functions.
+ */
+typedef uint32_t dif_otp_ctrl_irq_snapshot_t;
+
+/**
+ * A hardware-level status code.
+ */
+typedef enum dif_otp_ctrl_status {
+  /**
+   * Indicates an error occured in the `CreatorSwCfg` partition.
+   */
+  kDifOtpCtrlStatusCreatorSwCfgError,
+  /**
+   * Indicates an error occured in the `OwnerSwCfg` partition.
+   */
+  kDifOtpCtrlStatusOwnerSwCfgError,
+  /**
+   * Indicates an error occured in the `HwCfg` partition.
+   */
+  kDifOtpCtrlStatusHwCfgError,
+  /**
+   * Indicates an error occured in the `LifeCycle` partition.
+   */
+  kDifOtpCtrlStatusLifeCycleError,
+  /**
+   * Indicates an error occured in the `Secret0` partition.
+   */
+  kDifOtpCtrlStatusSecret0Error,
+  /**
+   * Indicates an error occured in the `Secret1` partition.
+   */
+  kDifOtpCtrlStatusSecret3Error,
+  /**
+   * Indicates an error occured in the `Secret2` partition.
+   */
+  kDifOtpCtrlStatusSecret2Error,
+
+  /**
+   * Indicates an error occured in the direct access interface.
+   */
+  kDifOtpCtrlStatusDaiError,
+  /**
+   * Indicates an error occured in the lifecycle interface.
+   */
+  kDifOtpCtrlStatusLciError,
+
+  /**
+   * Indicates that the direct access interface is idle.
+   */
+  kDifOtpCtrlStatusDaiIdle,
```

Comment:
A few more error bits are going to be added in this PR: https://github.com/lowRISC/opentitan/pull/3550